### PR TITLE
[CircleGraph] Implement visqselector mode

### PR DESCRIPTION
**Pull request by stamalakhov**
This commit implements visqselector mode and managing of scrollToSlected.

Its correctness is tested in https://github.com/Samsung/ONE-vscode/pull/1543

Fresh draft: https://github.com/Samsung/ONE-vscode/pull/1543
Full draft: https://github.com/Samsung/ONE-vscode/pull/1505
Related: https://github.com/Samsung/ONE-vscode/issues/1491

ONE-vscode-DCO-1.0-Signed-off-by: s.malakhov <s.malakhov@partner.samsung.com>

### CODE REVIEW

File: media/CircleGraph/view.js [hunk 1]
```diff
@@ -542,13 +542,24 @@ view.View = class {
...
+  }
+
   /**
    * @brief toggleSelect will select or toggle select with CtrlKey down
    * @param viewNode view.Node instance
    * @note  works on host mode is viewMode.selector
    */
   toggleSelect(viewNode) {
-    if (viewNode && this._host._mode === viewMode.selector) {
+    if (
```
Comment (dayo09) [inline media/CircleGraph/view.js:558]: How about returning 'if not' condition here to reduce the level of condition?
Like,
```
if (!viewMode) {return}
if (this._host._mode === viewMode.selector || this._host._mode === viewMode.visqselector) {return}

// continued logic
```

**Review by dayo09 — DISMISSED** [review]

File: media/CircleGraph/index.js [hunk 2]
```diff
@@ -52,6 +52,7 @@ const viewMode = {
   viewer: 0, // default circle viewer
   selector: 1, // circle partition editor node selector
   visq: 2, // quantization error viewer
+  visqselector: 3, // quantization error viewer which is able to select nodes
```
Comment (seanshpark) [inline media/CircleGraph/index.js:55]: plz refer https://github.com/Samsung/ONE-vscode/pull/1546#discussion_r1181994078

File: media/CircleGraph/view.js (hunk 1 above)
Comment (stamalakhov) [reply media/CircleGraph/view.js:558]: @dayo09 
Thank you. Unfortunately there are 4 values, including `visqselector`, so inverting logic will still produce:
```
if (this._host._mode !== viewMode.viewer && this._host._mode !== viewMode.visq) {return}

// continuous logic
```

File: media/CircleGraph/index.js [hunk 3]
```diff
@@ -52,6 +52,7 @@ const viewMode = {
...
   // refer https://github.com/Samsung/ONE-vscode/issues/1350
```
Comment (seanshpark) [inline media/CircleGraph/index.js:56]:
```suggestion
  // refer https://github.com/Samsung/ONE-vscode/issues/1350
  visqselector: 3, // quantization error viewer which is able to select nodes
```

Comment (seanshpark) [reply media/CircleGraph/index.js:56]: comment about 1350 is for `visq: 2`.

Comment (stamalakhov) [reply media/CircleGraph/index.js:56]: Ahh. Sorry. Thank you.

**Review by seanshpark — DISMISSED** [review]
LGTM

File: media/CircleGraph/view.js (hunk 1 above)
Comment (dayo09) [reply media/CircleGraph/view.js:558]: I mean, if it's not 'viewMode', then this function don't need to proceed. Note that it's an 'and' condition. 😄

Comment (stamalakhov) [reply media/CircleGraph/view.js:558]: Ahhh. Sorry. I didn't immediately understand that.

Comment (stamalakhov) [reply media/CircleGraph/view.js:561]: @dayo09 
fixed.

**Review by dayo09 — APPROVED** [review]
Thanks!

File: media/CircleGraph/view.js [hunk 4]
```diff
@@ -542,13 +542,26 @@ view.View = class {
...
   /**
    * @brief toggleSelect will select or toggle select with CtrlKey down
    * @param viewNode view.Node instance
    * @note  works on host mode is viewMode.selector
    */
   toggleSelect(viewNode) {
-    if (viewNode && this._host._mode === viewMode.selector) {
+    if (!viewNode) {
+      return;
+    }
```
Comment (seanshpark) [inline media/CircleGraph/view.js:560]: Please do not change like this. I don't like return in the middle of a method.

Comment (stamalakhov) [reply media/CircleGraph/view.js:560]: ok.

Comment (dayo09) [reply media/CircleGraph/view.js:560]: I don't mind. Let me know when it's ready :-D

Comment (stamalakhov) [reply media/CircleGraph/view.js]: @dayo09 @seanshpark 
fixed

**Review by dayo09 — APPROVED** [review]

File: media/CircleGraph/index.js (hunk 2 above)
Comment (seanshpark) [reply media/CircleGraph/index.js:55]: plz move this line below 1350 link comment

Comment (stamalakhov) [reply media/CircleGraph/index.js:55]: Ahhh. Sorry.

**Review by dayo09 — APPROVED** [review]

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**